src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -125,22 +125,23 @@ public function can($subject, $transitionName)
      */
     public function apply($subject, $transitionName)
     {
-        $transitions = $this->getEnabledTransitions($subject);
-
-        // We can shortcut the getMarking method in order to boost performance,
-        // since the "getEnabledTransitions" method already checks the Marking
-        // state
-        $marking = $this->markingStore->getMarking($subject);
-
-        $applied = false;
+        $marking = $this->getMarking($subject);
+        $transitions = [];
 
-        foreach ($transitions as $transition) {
-            if ($transitionName !== $transition->getName()) {
+        foreach ($this->definition->getTransitions() as $transition) {
+            if ($transition->getName() !== $transitionName) {
                 continue;
             }
+            if ($this->doCan($subject, $marking, $transition)) {
```

Comment:
@lyrixx 
what about having the same check as can()

https://github.com/symfony/symfony/blob/72cec555fd76c8c8284a575ff4156b9a959f660d/src/Symfony/Component/Workflow/Workflow.php#L107

and avoid the upper continue

Reply:
Fixed. Thanks